libs/vkd3d/swapchain.c

```diff
@@ -1573,6 +1661,15 @@ static void dxgi_vk_swap_chain_recreate_swapchain_in_present_task(struct dxgi_vk
     swapchain_create_info.imageExtent.height = max(swapchain_create_info.imageExtent.height, surface_caps.minImageExtent.height);
     swapchain_create_info.imageExtent.height = min(swapchain_create_info.imageExtent.height, surface_caps.maxImageExtent.height);
 
+    if (chain->queue->device->vk_info.NV_low_latency2)
+    {
+        memset(&swapchain_latency_create_info, 0, sizeof(swapchain_latency_create_info));
+        swapchain_latency_create_info.sType = VK_STRUCTURE_TYPE_SWAPCHAIN_LATENCY_CREATE_INFO_NV;
+        swapchain_latency_create_info.pNext = NULL;
+        swapchain_latency_create_info.latencyModeEnable = true;
+        swapchain_create_info.pNext = &swapchain_latency_create_info;
+    }
+
```

Comment:
Shouldn't the swapchain lock be held here? It's possible app functions can race with swapchain recreation, right?

Reply:
Yes I think that should be added. The call to vkCreateSwapchainKHR, and dxgi_vk_swap_chain_set_low_latency_state are now protected by the low latency swapchain lock.